flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -322,7 +341,7 @@ public int unsynchronizedGetNumberOfQueuedBuffers() {
 	public void flush() {
 		final boolean notifyDataAvailable;
 		synchronized (buffers) {
-			if (buffers.isEmpty()) {
+			if (buffers.isEmpty() || readView == null || isBlockedByCheckpoint) {
```

Comment:
I think we should not modify the condition of `flushRequested`, and only work on the condition of `notifyDataAvailable`.
Then `isBlockedByCheckpoint` can be added into below `notifyDataAvailable`, but now touch `readerView` condition which is out of scope of this PR.